pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -270,6 +294,44 @@ func (s *GameServerSelector) matchCounters(gs *agonesv1.GameServer) bool {
 	return true
 }
 
+// CounterActions attempts to peform any actions from the CounterAction on the GameServer Counter.
+// Returns a string list of any actions that could not be performed.
+func (ca *CounterAction) CounterActions(counter string, gs *agonesv1.GameServer) []string {
```

Comment:
I find the return of an array of strings a bit "non-Go"-ish, and also we're losing contextual information along the way.

A few options:

1) use errors.Wrap() and just keep wrapping errors (it's probably fine?)
2) If you want a custom formatting for error output, let's define a small multi-error type 
3) Use something like https://github.com/hashicorp/go-multierror
4) I like this least, but return an []err as at least you know it's an error (but I don't like this as much)

Reply:
Looks like https://pkg.go.dev/errors#Join would work here.

Reply:
Updated the Go version to 1.20 so we can use Join.